src/Symfony/Component/HttpKernel/Kernel.php

```diff
@@ -481,7 +492,21 @@ protected function initializeContainer()
         $cache = new ConfigCache($this->getCacheDir().'/'.$class.'.php', $this->debug);
         $fresh = true;
         if (!$cache->isFresh()) {
+            $serviceClass = $this->getServiceClass();
+            $bundles = array();
+            foreach ($this->bundles as $name => $bundle) {
+                $bundles[$name] = array(
+                    'service_class' => method_exists($bundle, 'getServiceClass') ? $bundle->getServiceClass() : __NAMESPACE__.'\\Service\Bundle',
+                    'class' => get_class($bundle),
+                    'namespace' => $bundle->getNamespace(),
+                    'parent' => $bundle->getParent(),
+                    'path' => $bundle->getPath(),
+                );
+            }
+            $serviceDefinition = new ServiceAwareDefinition($serviceClass, array($this->environment, $this->debug, $bundles));
+            $serviceDefinition->setService(new $serviceClass($this->environment, $this->debug, $bundles));
             $container = $this->buildContainer();
+            $container->setDefinition('kernel_as_a_service', $serviceDefinition);
```

Comment:
If you want to use this service in an extension, you'll still have to do quirky things in `MergeExtensionConfigurationPass` to make it available in the `$tmpContainer` passed to the `ExtensionInterface::load` method, right ?